go/roothash/api/commitment/pool_test.go

```diff
@@ -377,6 +377,49 @@ func TestPoolStragglers(t *testing.T) {
 		ddEc := dc.ToDDResult().(*ExecutorCommitment)
 		require.EqualValues(t, &ec, ddEc, "DD should return the correct commitment")
 	})
+
+	t.Run("StragglerFailure", func(t *testing.T) {
```

Comment:
Could also test if we switch to the discrepancy mode before the timer expires if there are too many failures,
```
	t.Run("TooManyStragglers", func(t *testing.T) {
		// Create a pool.
		pool := Pool{
			Runtime:   rt,
			Committee: committee,
			Round:     0,
		}

		// Generate a commitment.
		childBlk, _, ec := generateExecutorCommitment(t, pool.Round)

		ec1 := ec
		ec1.NodeID = sk1.Public()
		ec1.Header.SetFailure(FailureUnknown)
		err := ec1.Sign(sk1, rt.ID)
		require.NoError(t, err, "ec1.Sign")

		ec2 := ec
		ec2.NodeID = sk2.Public()
		ec2.Header.SetFailure(FailureUnknown)
		err = ec2.Sign(sk2, rt.ID)
		require.NoError(t, err, "ec2.Sign")

		// Add first commitment (indicates failure).
		err = pool.AddExecutorCommitment(context.Background(), childBlk, nl, &ec1, nil)
		require.NoError(t, err, "AddExecutorCommitment")

		// The number of failures is equal to the number of allowed stragglers.
		_, err = pool.ProcessCommitments(false)
		require.Error(t, err, "ProcessCommitments")
		require.Equal(t, ErrStillWaiting, err, "ProcessCommitments")

		// Add second commitment (indicates failure).
		err = pool.AddExecutorCommitment(context.Background(), childBlk, nl, &ec2, nil)
		require.NoError(t, err, "AddExecutorCommitment")

		// The number of failures exceeds the number of allowed stragglers.
		_, err = pool.ProcessCommitments(false)
		require.Equal(t, ErrDiscrepancyDetected, err, "ProcessCommitments")
	})
```